service/src/main/java/skills/services/ContactUsersService.groovy

```diff
@@ -37,12 +39,59 @@ import java.util.stream.Stream
 @Service
 class ContactUsersService {
 
+    private Parser parser = Parser.builder().build()
+    private HtmlRenderer renderer = HtmlRenderer.builder().build()
+
     @Autowired
     NativeQueriesRepo nativeQueriesRepo
 
     @Autowired
     EmailNotifier emailNotifier
 
+    @Autowired
+    AccessSettingsStorageService accessSettingsStorageService
+
+    @Transactional(readOnly = true)
+    Long countAllProjectAdminsWithEmail() {
+        return accessSettingsStorageService.countUserIdsWithRoleAndEmail(RoleName.ROLE_PROJECT_ADMIN)
+    }
+
+    @Transactional
+    contactAllProjectAdmins(String emailSubject, String emailBody) {
+        Parser parser = Parser.builder().build()
+        HtmlRenderer renderer = HtmlRenderer.builder().build()
+        def markdown = parser.parse(emailBody)
+        String parsedBody = renderer.render(markdown)
+
+        List<String> batch = []
+        Closure sendNotifications = { List<String> userIds ->
+            Notifier.NotificationRequest request = new Notifier.NotificationRequest(
+                    userIds: new ArrayList(batch),
+                    type: Notification.Type.ContactUsers,
+                    keyValParams: [
+                            htmlBody    : parsedBody,
+                            emailSubject: emailSubject,
+                            rawBody     : emailBody,
+                    ]
+            )
+            emailNotifier.sendNotification(request)
+            batch.clear()
+        }
+        getAllProjectAdminsWithEmail().withCloseable { Stream<String> userIds ->
+            userIds.forEach {
+                batch.add(it)
+                if (batch.size() == 150) {
```

Comment:
suggesting to make this batch size configurable; set it low enough in the tests to exercise sending multiple notifications